src/sss_client/common.c

```diff
@@ -767,6 +767,15 @@ static enum sss_status sss_cli_check_socket(int *errnop,
         myself_ino = myself_sb.st_ino;
     }
 
+    /* check if the socket has beed was hijacked */
+    if (sss_cli_sd_get() != -1) {
+        ret = fstat(sss_cli_sd_get(), &mypid_sb);
+        if ((ret != 0) || (!S_ISSOCK(mypid_sb.st_mode))
+            || (mypid_sb.st_ino != sss_cli_sb->st_ino)) {
```

Comment:
since inodes are local to a given device, when comparing inodes between two files its usually a good idea to compare st_dev too

Reply:
+1

Reply:
Fixed here and in `sss_nss_mc_validate()`